src/bufr/BufrParser/Query/QueryParser.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * (C) Copyright 2022 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "QueryParser.h"
+
+#include <iostream>
+#include <algorithm>
+
+namespace Ingester {
+namespace bufr {
+     std::vector<std::string> QueryParser::splitMultiquery(const std::string &query) {
+        std::vector<std::string> subqueries;
+
+        // Remove whitespace from query and assign to working_str
+        std::string working_str = query;
+        working_str.erase(
+                std::remove(working_str.begin(), working_str.end(), ' '), working_str.end());
+
+        if (working_str.substr(0, 1) == "[") {
+            if (working_str.substr(working_str.length() - 1, 1) != "]") {
+                std::cerr << "Query Parser: multi query is lacking closing brackets." << std::endl;
+                exit(1);
+            }
+
+            working_str = working_str.substr(1, working_str.length() - 2);
+
+            std::vector<size_t> comma_positions;
```

Comment:
What's the benefit of using a "vectors" in a lot of these?

Reply:
Vectors are equivalent to Lists in python. You use them basically the same way. So every time you see a vector its just a List.